esdt-safe/src/to_sovereign/create_tx.rs

```diff
@@ -14,29 +14,41 @@ pub trait CreateTxModule:
     + max_bridged_amount_module::MaxBridgedAmountModule
     + multiversx_sc_modules::pause::PauseModule
 {
-    /// Create an Elrond -> Sovereign transaction.
+    /// Create an MultiversX -> Sovereign transaction.
     #[payable("*")]
     #[endpoint]
     fn deposit(
         &self,
         to: ManagedAddress,
-        opt_transfer_data: OptionalValue<TransferData<Self::Api>>,
+        opt_gas_limit: OptionalValue<GasLimit>,
+        opt_function: OptionalValue<ManagedBuffer>,
```

Comment:
too many optionals - it is a bad practice. rewrite this. The thing is if we have gasLimit - we must have a function and we might have opt_args.

So make optional a structure which will have the GasLimit, Function and ArgsList.